From 4fe8ad298f9d859fea9f619d2de7ad8c5c2428ac Mon Sep 17 00:00:00 2001
From: Nikolay Borisov <nborisov@suse.com>
Date: Fri, 9 Jun 2017 15:31:13 +0300
Subject: [PATCH] kABI: protect struct xlog
Patch-mainline: never, kabi
References: bsc#1043598

Commit 12818d24db8a ("xfs: rework log recovery to submit buffers on LSN boundaries")
added a member at the end of struxt xlog. Since it's internal and dynamically
allocated in-memory struct just protect it with __GENKSYMS__

Signed-off-by: Nikolay Borisov <nborisov@suse.com>
---
 fs/xfs/xfs_log_priv.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/xfs/xfs_log_priv.h b/fs/xfs/xfs_log_priv.h
index 1f5fada95e7a..958ed15a8899 100644
--- a/fs/xfs/xfs_log_priv.h
+++ b/fs/xfs/xfs_log_priv.h
@@ -411,8 +411,10 @@ struct xlog {
 #ifdef DEBUG
 	void			*l_iclog_bak[XLOG_MAX_ICLOGS];
 #endif
+#ifndef __GENKSYMS__
 	/* log recovery lsn tracking (for buffer submission */
 	xfs_lsn_t		l_recovery_lsn;
+#endif
 };
 
 #define XLOG_BUF_CANCEL_BUCKET(log, blkno) \
-- 
2.7.4

